src/variables/ShellCommandVariable_CaretPosition.ts

```diff
@@ -0,0 +1,98 @@
+import {getEditor} from "../Common";
+import {addShellCommandVariableInstructions} from "./ShellCommandVariableInstructions";
+import {IParameters, ShellCommandVariable} from "./ShellCommandVariable";
+import {IAutocompleteItem} from "../settings/setting_elements/Autocomplete";
+
+export class ShellCommandVariable_CaretPosition extends ShellCommandVariable {
+    static variable_name = "caret_position";
+    static help_text = "Gives the line number and column position of the current caret position as 'line:col'. Get only the line number using {{caret_position:line}}, and only the column with {{caret_position:column}}. Line and column numbers are 1-indexed.";
+
+    protected static readonly parameters: IParameters = {
+        mode: {
+            options: ["line", "column"],
+            required: false,
+        },
+    };
+
+    protected arguments: {
+        mode: string;
+    }
+
+    generateValue(): string {
+        // Check that we are able to get an editor
+        let editor = getEditor(this.app);
+        if (null === editor) {
+            // Nope.
+            this.newErrorMessage("Could not get an editor instance! Please raise an issue in GitHub.");
+            return null;
+        }
+
+        let position = editor.getCursor('to');
```

Comment:
I think the variable will not be changed, so it can be a constant?
```suggestion
        const position = editor.getCursor('to');
```